core/src/main/java/org/mule/runtime/core/processor/AsyncInterceptingMessageProcessor.java

```diff
@@ -104,13 +108,13 @@ protected Event processNextTimed(Event event) throws MuleException {
 
   protected boolean isProcessAsync(Event event) throws MuleException {
     if (!canProcessAsync(event)) {
-      throw new DefaultMuleException(createStaticMessage(SYNCHRONOUS_NONBLOCKING_EVENT_ERROR_MESSAGE));
+      throw new DefaultMuleException(createStaticMessage(SYNCHRONOUS_EVENT_ERROR_MESSAGE));
     }
     return doThreading && canProcessAsync(event);
   }
 
   protected boolean canProcessAsync(Event event) {
-    return !(event.isSynchronous() || event.isTransacted() || event.getReplyToHandler() instanceof NonBlockingReplyToHandler);
+    return !(event.isSynchronous() || event.isTransacted());
```

Comment:
It may go, don't want to fill code with TODO for things which are sequential work items, rather for separate issues that need doing.